util/cpu/cpu.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cpu
+
+import (
+	"os"
+	"sync"
+	"time"
+
+	"github.com/cloudfoundry/gosigar"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/util/cgroup"
+	"github.com/pingcap/tidb/util/mathutil"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+)
+
+var cpuUsage atomic.Float64
+
+// GetCPUUsage returns the cpu usage of the current process.
+func GetCPUUsage() float64 {
+	return cpuUsage.Load()
+}
+
+// Observer is used to observe the cpu usage of the current process.
+type Observer struct {
+	utime int64
+	stime int64
+	now   int64
+	exit  chan struct{}
```

Comment:
What is this exit means？

Reply:
CPU Observer will create a gorotine to collect the cpu monitor info. When TiDB exit, we need to notice this gorotine to exit.

Reply:
This CPU Observer goroutine is not in TiDB process？

Reply:
In TiDB process, it will be started by resource manage.